tests/zfs-tests/tests/functional/cli_root/zfs_load-key/zfs_load-key_tty.ksh

```diff
@@ -0,0 +1,57 @@
+#!/bin/ksh -p
+#
+# CDDL HEADER START
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright (c) 2020 Adam Moss, Inc. All rights reserved.
+#
+
+. $STF_SUITE/include/libtest.shlib
+. $STF_SUITE/tests/functional/cli_root/zfs_load-key/zfs_load-key_common.kshlib
+
+#
+# DESCRIPTION:
+# 'zfs load-key' accepts the key from tty
+#
+# STRATEGY:
+# 1. Create an encrypted dataset
+# 2. Attempt to load the dataset's key from a fake tty
+# 3. Verify the key is loaded
+#
+
+verify_runnable "both"
+
+function cleanup
+{
+	datasetexists $TESTPOOL/$TESTFS1 && \
+		log_must zfs destroy $TESTPOOL/$TESTFS1
+}
+log_onexit cleanup
+
+log_assert "'zfs load-key' accepts the key from tty"
+
+# 1.
+log_must eval "echo $PASSPHRASE | zfs create -o encryption=on" \
+	"-o keyformat=passphrase -o keylocation=prompt $TESTPOOL/$TESTFS1"
+
+# 2.
+log_must zfs unmount $TESTPOOL/$TESTFS1
+log_must zfs unload-key $TESTPOOL/$TESTFS1
+log_must faketty "echo $PASSPHRASE | zfs load-key $TESTPOOL/$TESTFS1"
```

Comment:
I'm not sure why, but this doesn't appear to work on the FreeBSD bots.

```
13:08:37.97 NOTE: Performing local cleanup via log_onexit (cleanup)
13:08:38.07 SUCCESS: zfs destroy -f testpool/testfs
Test: /usr/local/share/zfs/zfs-tests/tests/functional/cli_root/zfs_load-key/zfs_load-key_tty (run as root) [00:00] [FAIL]
13:26:38.51 ASSERTION: 'zfs load-key' accepts the key from tty
13:26:38.87 SUCCESS: eval echo password | zfs create -o encryption=on -o keyformat=passphrase -o keylocation=prompt testpool/testfs1
13:26:38.91 SUCCESS: zfs unmount testpool/testfs1
13:26:38.92 SUCCESS: zfs unload-key testpool/testfs1
13:26:38.93 ^Denv: echo password | zfs load-key testpool/testfs1: No such file or directory
13:26:38.93 
13:26:38.93 ERROR: faketty echo password | zfs load-key testpool/testfs1 exited 127
13:26:38.93 NOTE: Performing test-fail callback (/usr/local/share/zfs/zfs-tests/callbacks/zfs_dmesg.ksh)
```

Reply:
Try `log_must faketty sh -c " echo $PASSPHRASE | zfs load-key $TESTPOOL/$TESTFS1"`
faketty does `env "$@"` on FreeBSD in order to handle setting env vars in the command.